src/ctrans.c

```diff
@@ -475,155 +656,106 @@ unsigned long c_grid3d(double *dout, unsigned long *nout, double *standarderror,
       pos += grid_pos[1] * n_grid[2];
       pos += grid_pos[2];
 
+
       // Store the answer
       dout[pos] = dout[pos] + *data_ptr;
       nout[pos] = nout[pos] + 1;
 
       // Calculate the standard deviation quantities
 
-      if(standarderror){
-      	if(nout[pos] == 1){
-      	  Mk[pos] = *data_ptr;
-      	  Qk[pos] = 0.0;
-      	} else {
-      	  Qk[pos] = Qk[pos] + ((nout[pos] - 1) * pow(*data_ptr - Mk[pos],2) / nout[pos]);
-      	  Mk[pos] = Mk[pos] + ((*data_ptr - Mk[pos]) / nout[pos]);
-      	}
-      }
+      Qk[pos] = Qk[pos] + ((nout[pos] - 1) * pow(*data_ptr - Mk[pos],2) / nout[pos]);
+      Mk[pos] = Mk[pos] + ((*data_ptr - Mk[pos]) / nout[pos]);
+      //fprintf(stderr, "Qk = %f, Mk = %f\n", Qk[pos], Mk[pos]);
 
       // Increment pointer
       data_ptr++;
     } else {
-      n_outside++;
+      data->n_outside++;
       data_ptr+=2;
     }
   }
-  
-  // Calculate mean by dividing by the number of data points in each
-  // voxel
 
-  if(norm_data){
-    for(i = 0; i < grid_size; i++){
-      if(nout[i] > 0){
-	       dout[i] = dout[i] / nout[i];
-      } else {
-	       dout[i] = 0.0;
-      }
-    }
-  }
+  return NULL;
+}
 
-  // Calculate the sterror
-  
-  if(standarderror){
-    for(i=0;i<grid_size;i++){
-      if(nout[i] > 1){
-      	// standard deviation of the sample distribution
-      	standarderror[i] = pow(Qk[pos] / (nout[i] - 1), 0.5) / pow(nout[i], 0.5);
-      }
-    }
-  }
+long nproc(void) {
+  long _n;
+#ifdef USE_THREADS 
 
-  if(Mk){
-    free(Mk);
+  _n = sysconf(_SC_NPROCESSORS_ONLN);
+  if(_n > MAX_THREADS){
+    _n = MAX_THREADS;
   }
-  if(Qk){
-    free(Qk);
-  }
-	
-  return n_outside;
-}
 
-// // This is the python 2 version
-// PyMODINIT_FUNC initctrans(void)  {
-// 	(void) Py_InitModule3("ctrans", _ctransMethods, _ctransDoc);
-// 	import_array();  // Must be present for NumPy.  Called first after above line.
-// }
-struct module_state {
-    PyObject *error;
-};
-
-#if PY_MAJOR_VERSION >= 3
-#define GETSTATE(m) ((struct module_state*)PyModule_GetState(m))
 #else
-#define GETSTATE(m) (&_state)
-static struct module_state _state;
+
+  _n = 1;
+
 #endif
 
-static PyObject *
-error_out(PyObject *m) {
-    struct module_state *st = GETSTATE(m);
-    PyErr_SetString(st->error, "something bad happened");
-    return NULL;
+  return _n;
 }
 
-static PyMethodDef _ctransMethods[] = {
-  {NULL, NULL, 0, NULL}     /* Sentinel - marks the end of this structure */
-};
+static PyObject* get_threads(PyObject *self, PyObject *args){
+  return PyLong_FromLong((long)_n_threads);
+}
+
+static PyObject* set_threads(PyObject *self, PyObject *args){
+  int threads;
+  if(!PyArg_ParseTuple(args, "i", &threads)){
+    return NULL;
+  }
+  if(threads > MAX_THREADS){
+    PyErr_SetString(PyExc_ValueError, "Requested number of threads > MAX_THREADS");
+    return NULL;
+  }
+  _n_threads = threads;
```

Comment:
This should be protected with `#ifdef USE_THREAD` to only allow `_n_threads` to be 1 if threading is not being used.